### PR TITLE
exodus-android-app: Upstream gradle, libraries and code

### PR DESCRIPTION
**Pull request by theimpulson**
This pull request aims to:

- Upstream Gradle and libraries to the latest stable upstream version
- Upstream deprecated code as much as possible

Commit messages are self-explanatory on what changes have been made.

### CODE REVIEW

**Comment by theimpulson** [comment]
Marked as draft as I intend to upstream more code. Suggestions are welcome in case any modifications will be required to merge this PR.

**Comment by theimpulson** [comment]
Ready from my side

File: app/src/main/java/org/eu/exodus_privacy/exodusprivacy/fragments/TrackerFragment.java [hunk 1]
```diff
@@ -58,7 +59,11 @@ public void onUpdateComplete() {
         trackerBinding.name.setText(tracker.name);
         trackerBinding.codeDetection.setText(tracker.codeSignature);
         trackerBinding.networkDetection.setText(tracker.networkSignature);
-        trackerBinding.description.setText(Html.fromHtml(Utils.markdownToHtml(tracker.description)));
+        if (Build.VERSION.SDK_INT > Build.VERSION_CODES.N) {
```
Comment (mypplication) [inline app/src/main/java/org/eu/exodus_privacy/exodusprivacy/fragments/TrackerFragment.java:62]: You can use HtmlCompat.fromHtml(..)

**Review by mypplication — APPROVED** [review]
Consider also update .travis.yml file?

File: app/src/main/java/org/eu/exodus_privacy/exodusprivacy/fragments/TrackerFragment.java (hunk 1 above)
Comment (theimpulson) [reply app/src/main/java/org/eu/exodus_privacy/exodusprivacy/fragments/TrackerFragment.java]: Done, thanks for the hint.

**Comment by theimpulson** [comment]
> Consider also update .travis.yml file?

Unfortunately, I cannot test any changes done in the CI file as Travis-CI seems to be asking me to choose a plan (probably pricing changed since I last used it). The only major change required seems to be simply moving to `openjdk11` for the jdk.

**Comment by pnu-s** [comment]
Thanks @theimpulson for the proposed change, I assigned the review to some people who, I hope, can do it.

About the `.travis.yml` change, AFAIK there were some changes in Travis indeed and we tried to move to GitHub Actions for all our repositories. Should probably do the same change here.

I'll open an issue for that purpose.

**Review by jfoucry — APPROVED** [review]

**Comment by theimpulson** [comment]
```
[2022-01-03 06:38:37] [build-stdout] [2022-01-03 06:38:37] [autobuild] Caused by: com.android.builder.errors.EvalIssueException: Android Gradle plugin requires Java 11 to run. You are currently using Java 1.8.
```

**Comment by pnu-s** [comment]
@theimpulson Thanks for trying to solve the issue with LGTM :)

It looks like the file should be named `lgtm.yml` instead of `.lgtm.yml`, could it explain why your last change did not fix it?

**Comment by theimpulson** [comment]
> @theimpulson Thanks for trying to solve the issue with LGTM :)
> 
> It looks like the file should be named `lgtm.yml` instead of `.lgtm.yml`, could it explain why your last change did not fix it?

@pnu-s there is a new error now which is due to LGTM not having Android SDK 31 tools present.

```
[2022-01-03 06:48:39] [build-stdout] [2022-01-03 06:48:39] [autobuild] To honour the JVM settings for this build a single-use Daemon process will be forked. See https://docs.gradle.org/7.0.2/userguide/gradle_daemon.html#sec:disabling_the_daemon.
[2022-01-03 06:48:41] [build-stdout] [2022-01-03 06:48:41] [autobuild] Daemon will be stopped at the end of the build 
[2022-01-03 06:48:54] [build-stdout] [2022-01-03 06:48:54] [autobuild] Checking the license for package Android SDK Platform 31 in /opt/extra/android/versions/latest/licenses
[2022-01-03 06:48:54] [build-stdout] [2022-01-03 06:48:54] [autobuild] License for package Android SDK Platform 31 accepted.
[2022-01-03 06:48:54] [build-stdout] [2022-01-03 06:48:54] [autobuild] Preparing "Install Android SDK Platform 31 (revision: 1)".
[2022-01-03 06:48:55] [build-stdout] [2022-01-03 06:48:54] [autobuild] Warning: Failed to read or create install properties file.
[2022-01-03 06:48:55] [build-stdout] [2022-01-03 06:48:55] [autobuild] FAILURE: Build failed with an exception.
[2022-01-03 06:48:55] [build-stdout] [2022-01-03 06:48:55] [autobuild] * What went wrong:
[2022-01-03 06:48:55] [build-stdout] [2022-01-03 06:48:55] [autobuild] Could not determine the dependencies of task ':app:processFedilab_fdroidReleaseResources'.
[2022-01-03 06:48:55] [build-stdout] [2022-01-03 06:48:55] [autobuild] > Failed to install the following SDK components:
[2022-01-03 06:48:55] [build-stdout] [2022-01-03 06:48:55] [autobuild]       platforms;android-31 Android SDK Platform 31
[2022-01-03 06:48:55] [build-stdout] [2022-01-03 06:48:55] [autobuild]   The SDK directory is not writable (/opt/extra/android/versions/latest)
[2022-01-03 06:48:55] [build-stdout] [2022-01-03 06:48:55] [autobuild] * Try:
[2022-01-03 06:48:55] [build-stdout] [2022-01-03 06:48:55] [autobuild] Run with --info or --debug option to get more log output. Run with --scan to get full insights.
```

I came across a similar issue here: https://github.com/github/securitylab/discussions/226. As per that it seems the LGTM is responsible for making the new versions of SDK available on their platform.

**Comment by theimpulson** [comment]
@pnu-s is using LGTM necessary? I was thinking of opening issues for improvement in the project among which was:

- Migrating to Kotlin,
- Using [ktlint](https://github.com/pinterest/ktlint) and standard android lint checks in actions,
- Add some tests,
- Uploading artifacts for testing using [upload-a-build-artifact](https://github.com/marketplace/actions/upload-a-build-artifact)
 
All of these should be more than enough to improve and maintain better code quality and builds.

**Comment by pnu-s** [comment]
Sounds sensible to me.

LGTM was added as an easy first step, but it should not block our PRs for these technical problems.
I'll merge this one as is then! :rocket:

**Pull request merged**